test-runner/wasi_test_runner/test_case.py

```diff
@@ -63,5 +70,5 @@ def _validate_dict(cls: Type[T], dict_config: Dict[str, Any]) -> None:
 class TestCase(NamedTuple):
     name: str
     config: Config
-    result: Result
+    result: Union[Result | SkippedResult | TimedoutResult]
```

Comment:
I've never seen this syntax before; IIRC it should be either `Union[Result, SkippedResult, TimedoutResult]` or `Result | SkippedResult | TimedoutResult`, see https://docs.python.org/3/library/typing.html#typing.Union